server/consumer.go

```diff
@@ -48,7 +48,9 @@ type ConsumerInfo struct {
 }
 
 type ConsumerConfig struct {
+	// Durable is deprecated. All consumers will have names. picked by clients.
 	Durable         string          `json:"durable_name,omitempty"`
+	Name            string          `json:"name,omitempty"`
```

Comment:
These kinds of depreations add ambiguity without much value, in effect in future all consuers will be durable.  So the existing key name is fine and accurate.  More so because its essentially an internal only detail hidden from users.

Not changing has the benefit of not adding weird changes or impacting systems that are based on remediation loops like terraform/k8s which would then see this as a change which will need remediation and would require quite careful specific handling in each of those systemsetc.  Or older client versions that does strict config compares etc. 

The cost/benefit of this change seems off when the existing name is correct already, even in future.

Etherway, this close to 2.9 this seems risky when lookin at the related changes overall vs the stated goal of this PR to me.

Reply:
Not all consumers will need to be durable, they will have a Name and optional InactiveThreshold, and Durable will be phased out.

This was needed for JS simplification as well.

Reply:
A durable wirh a timeout would survive restarts before the timeout? So it’s a durable. 

Chafing this key will complicate things like terraform and CRDs and doesn’t bring any functionality. It’s cosmetic at best no effect here but big knock on elsewhere.

Reply:
An ephemeral that is backed by disk with an Inactive timeout would as well.

Reply:
So then changing the key is just not needed. The behaviour is always durable with an optional idle timeout. 

Durable is a fine name and doesn’t break other things.

Reply:
Durable meant no cleanup, and we had the notion of Name when you asked for consumer info which put together durable and ephemerals.

This is the right direction.